libs/deal-ticket/src/components/deal-ticket/margin-mode-selector.tsx

```diff
@@ -0,0 +1,246 @@
+      setLeverage(max);
+    }
+  }, [max, maxLeverage, leverage]);
+
+  const t = useT();
+  return (
+    <Dialog
+      title={t('Isolated margin')}
+      size="small"
+      open={open}
+      onChange={(isOpen) => {
+        if (!isOpen) {
+          onClose();
+        }
+      }}
+    >
+      <div className="text-sm mb-4">
+        <p className="mb-1">
+          {t('You are setting this market to isolated margin mode.')}
+        </p>
+        <p className="mb-1">
+          {t(
+            'Set the leverage you want below. The maximum leverage you can take is determined by the risk model of the market.'
+          )}
+        </p>
+        <p className="mb-1">
+          {t(
+            'Only your allocated margin will be used to fund this position, and if the maintenance margin is breached you will be closed out.'
+          )}
+        </p>
+      </div>
+      <form
+        onSubmit={() => {
+          create({
+            updateMarginMode: {
+              market_id: marketId,
+              mode: MarginMode.MARGIN_MODE_ISOLATED_MARGIN,
+              marginFactor: `${1 / leverage}`,
+            },
+          });
+          onClose();
+        }}
+      >
+        <FormGroup label={t('Leverage')} labelFor="leverage-input" compact>
+          <div className="mb-2">
+            <LeverageSlider
+              max={max}
+              step={0.1}
+              value={[leverage]}
+              onValueChange={([value]) => setLeverage(value)}
+            />
+          </div>
+          <Input
+            type="number"
+            id="leverage-input"
+            min={1}
+            max={max}
+            step={0.1}
+            value={leverage}
+            onChange={(e) => setLeverage(Number(e.target.value))}
+          />
+        </FormGroup>
+        <Button className="w-full" type="submit">
+          {t('Confirm')}
+        </Button>
+      </form>
+    </Dialog>
+  );
+};
+
+export const MarginModeSelector = ({ marketId }: { marketId: string }) => {
+  const t = useT();
+  const [dialog, setDialog] = useState<'cross' | 'isolated' | ''>();
+  const { pubKey: partyId, isReadOnly } = useVegaWallet();
+  const { data: margin } = useDataProvider({
+    dataProvider: marketMarginDataProvider,
+    variables: {
+      partyId: partyId || '',
+      marketId,
+    },
+    skip: !partyId,
+  });
+  useEffect(() => {
+    if (!partyId) {
+      setDialog('');
+    }
+  }, [partyId]);
+  const create = useVegaTransactionStore((state) => state.create);
+  const marginMode = margin?.marginMode;
+  const marginFactor =
+    margin?.marginFactor && margin?.marginFactor !== '0'
+      ? margin?.marginFactor
+      : undefined;
+  const disabled = isReadOnly;
+  const onClose = () => setDialog(undefined);
+  const enabledModeClassName = 'bg-vega-clight-500 dark:bg-vega-cdark-500';
+
+  return (
+    <>
+      <div className="mb-4 grid h-8 leading-8 font-alpha text-xs grid-cols-2">
+        <button
```

Comment:
@daro-maj  @bglownia  These buttons feel a little broken when your wallet isn't connected. Perhaps lets change them to `cursor-default` wrap them with a tooltip prompting to connect?

Reply:
@mattrussell36 if it comes to the fact that the cross and isolated buttons are not active, we drew attention to it earlier, but the question is whether we need additional tooltip since we have a deal ticket stating that a wallet needs to be connected ?

Reply:
"Place order" button works in the same way. You can click but it does not do anything, until wallet is not connected. We should find consistent solution for whole deal ticket.